testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jca/deployment/JndiViewOperationJMSTestCase.java

```diff
@@ -0,0 +1,120 @@
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.jboss.as.test.integration.jca.deployment;
+
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.OP;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.OP_ADDR;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.SUBSYSTEM;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.as.test.integration.jca.JcaMgmtBase;
+import org.jboss.as.test.integration.jca.rar.MultipleAdminObject1;
+import org.jboss.dmr.ModelNode;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.jboss.shrinkwrap.api.spec.ResourceAdapterArchive;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * Test the situation when connection factory or JMS destination is configured for resource adapter
+ *
+ * @author <a href="parsharma@redhat.com">Parul Sharma</a>
+ */
+
+@RunWith(Arquillian.class)
+@RunAsClient
+public class JndiViewOperationJMSTestCase extends JcaMgmtBase {
+
+    static final String rarDeploymentName = "eis.rar";
+    private static ModelNode address;
+
+    @Deployment(name = "rar", order = 1)
+    public static Archive<?> deploytRar() {
+        ResourceAdapterArchive raa = ShrinkWrap.create(ResourceAdapterArchive.class, rarDeploymentName);
+
+        JavaArchive ja = ShrinkWrap.create(JavaArchive.class, "lib.jar");
+        ja.addPackage(MultipleAdminObject1.class.getPackage());
+        raa.addAsLibrary(ja);
+
+        raa.addAsManifestResource(JndiViewOperationJMSTestCase.class.getPackage(), "ra.xml", "ra.xml");
+        return raa;
+    }
+
+    @Test
+    public void testValueOfJms() throws Exception {
+        setup();
+        ModelNode address = getAddress();
+        ModelNode operation = getOperation(address);
+        ModelNode result = executeOperation(operation);
+
+         ModelNode value = result.get("java: contexts", "java:jboss", "Name3", "class-name");
+         Assert.assertEquals("org.jboss.as.test.integration.jca.rar.MultipleAdminObject1Impl", value.asString());
+
+    }
+
+    private ModelNode getAddress() {
+        return new ModelNode().add(SUBSYSTEM, "naming");
+    }
+
+    private ModelNode getOperation(ModelNode address) {
+        ModelNode operation = new ModelNode();
+        operation.get(OP).set("jndi-view");
+        operation.get(OP_ADDR).set(address);
+        return operation;
+    }
+
+    private void setup() throws Exception {
```

Comment:
I think deployment, undeployment and clean up is done by arquillian.

Reply:
@parsharma The configuration done in `@Test` method won't be clean up by Arquillian. The deployment archive with `@Deployment` will be undeployed by Arquillian. So it should be cleaned. It can be done either by cleaning it within your `@Test` method, or you can use `SnapshotRestoreSetupTask` for the configuration, you can refer to [TracerEnableTestCase#TracerEnableSetup](https://github.com/wildfly/wildfly/blob/master/testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jca/tracer/TracerEnableTestCase.java#L91) on the 2nd option.

Reply:
@gaol yes, yesterday I discussed with @yersan, he also suggested same by rolling back changes in the @test method itself. I referred to the same class you have suggested RarInsideEarReDeploymentTestCase. They are also removing address in finally, incase test class fails it will rollback all the changes.